### PR TITLE
update src references

### PR DESCRIPTION
**Pull request by mounika-nalla**
### Description

automation script to update src references for 
GH issue: https://github.ibm.com/GoldenEye/issues/issues/3523 
https://github.ibm.com/GoldenEye/issues/issues/3522

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] The PR references a GitHub issue.
- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Review by daniel-butler-irl — CHANGES_REQUESTED** [review]
Can we get a readme, with usage documentation? Also needs a `requirements.txt` with the 3rd party deps (git, requests)

File: scripts/run-update-src.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+import git
+import requests
+
+###########################################################################################################
+# Constants
+###########################################################################################################
+
+# text to be searched in files
+search_pattern_tim = (
```
Comment (daniel-butler-irl) [inline scripts/run-update-src.py:17]: Best practice in Python is to write constant variable names in `ALL_CAPS`

File: scripts/run-update-src.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                    data = replace_source(
+                        file, search_pattern_ge, replace_text, True, store
+                    )
+                    if data != "":
+                        write_modified_content(file, data)
+                print("Source references are updated in tf files")
+            else:
+                print("No md files found to update")
+
+
```
Comment (daniel-butler-irl) [inline scripts/run-update-src.py:209]: Put inside `if __name__ == "__main__":` as best practice. See https://realpython.com/if-name-main-python/

File: scripts/run-update-src.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+# get the current working repo
+def get_repo():
+    repo = git.Repo(os.getcwd())
+    repo_name = repo.remotes.origin.url
+    return repo_name
+
+
+# get source information from terraform registry
+def get_response(repo_name):
```
Comment (daniel-butler-irl) [inline scripts/run-update-src.py:45]: Please add type hinting and docstrings to all functions, for an example see https://github.ibm.com/GoldenEye/bootstrap-s3-backend-module/blob/master/scripts/switch_backend.py#L13
https://realpython.com/lessons/type-hinting/
https://realpython.com/documenting-python-code/#docstring-types

File: scripts/run-update-src.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+
+###########################################################################################################
+# Main
+###########################################################################################################
+
+
+def main():
+    for current_file_extension in files_to_be_searched:
+        if current_file_extension == "**/*.md":
```
Comment (daniel-butler-irl) [inline scripts/run-update-src.py:165]: As far as I can tell this block is identical to the if tf block, except the replace_source version_update parameter, is different. Generally, this is an indicator that the code needs to be extracted into a function. However in this case you can update to something similar:
```python
for current_file_extension in files_to_be_searched:
   vesion_update = None
   if current_file_extension == "**/*.md":
      version_update = True
   else if current_file_extension == "**/*.tf":
      version_update = False
   files = []
   store = []
   files, lines = get_files(current_file_extension, search_pattern_tim, [], [])
   # etc....
        data = replace_source(
                        file, search_pattern_ge, replace_text, version_update, store
                    )
   # etc....
   
```

File: scripts/run-update-src.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+    return store
+
+
+###########################################################################################################
+# Main
+###########################################################################################################
+
+
+def main():
+    for current_file_extension in files_to_be_searched:
```
Comment (daniel-butler-irl) [inline scripts/run-update-src.py:164]: I think a better name for this would be `current_file_pattern` as it better describes the value

**Review by ocofaigh — CHANGES_REQUESTED** [review]
We may in the future have other scripts that we may wish to add to the scripts folder. Suggest you create another folder within the scripts folder to place all these files in?

File: scripts/run-update-src.py [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+    search_pattern: str,
+    replace_text: str,
+    version_update: bool,
+    store: list,
+) -> str:
+    """Updates source information in the given files to the one in terrafrom-ibm-modules source
+
+    Args:
+       file(str): name of the file to be updated
+       search_pattern(str): search for GoldenEye or terraform-ibm-modules source links
```
Comment (daniel-butler-irl) [inline scripts/run-update-src.py:131]: again the function is not specific to goldeneye or terraform-ibm-modules

File: scripts/run-update-src.py [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+    Returns:
+    """
+
+    with open(file, "w") as f:
+        f.write(data)
+        f.close()
+
+
+# extract the repo name
+def extract_repo_name(repo_name: str) -> str:
```
Comment (daniel-butler-irl) [inline scripts/run-update-src.py:174]: you could update this to `def extract_repo_name(repo_name: str, prefix="terraform-ibm-":str) ->str` That way its more flexible

**Review by daniel-butler-irl — CHANGES_REQUESTED** [review]
Ok big improvement, just a few more changes

**Review by daniel-butler-irl — APPROVED** [review]

**Pull request merged**